npm/npm_import.bzl

```diff
@@ -30,23 +30,243 @@ load("//npm/private:versions.bzl", "PNPM_VERSIONS")
+            as well as a package version, then the version-specific args will be appended to the args for the package.
+
+        custom_postinstalls: A map of package names or package names with their version (e.g., "my-package" or "my-package@v1.2.3")
+            to a custom postinstall script to apply to the downloaded npm package after its lifecycle scripts runs.
+            If the version is left out of the package name, the script will run on every version of the npm package. If
+            a custom postinstall scripts exists for a package as well as for a specific version, the script for the versioned package
+            will be appended with `&&` to the non-versioned package script.
+
+        prod: If true, only install dependencies.
+
+        public_hoist_packages: A map of package names or package names with their version (e.g., "my-package" or "my-package@v1.2.3")
+            to a list of Bazel packages in which to hoist the package to the top-level of the node_modules tree when linking.
+
+            This is similar to setting https://pnpm.io/npmrc#public-hoist-pattern in an .npmrc file outside of Bazel, however,
+            wild-cards are not yet supported and npm_translate_lock will fail if there are multiple versions of a package that
+            are to be hoisted.
+
+        dev: If true, only install devDependencies
+
+        no_optional: If true, optionalDependencies are not installed
+
+        lifecycle_hooks_exclude: A list of package names or package names with their version (e.g., "my-package" or "my-package@v1.2.3")
+            to not run lifecycle hooks on
+
+        run_lifecycle_hooks: If true, runs preinstall, install and postinstall lifecycle hooks on npm packages if they exist
+
+        lifecycle_hooks_envs: Environment variables applied to the preinstall, install and postinstall lifecycle hooks on npm packages.
+            The environment variables can be defined per package by package name or globally using "*".
+            Variables are declared as key/value pairs of the form "key=value".
+
+            For example:
+
+            ```
+            lifecycle_hooks_envs: {
+                "*": ["GLOBAL_KEY1=value1", "GLOBAL_KEY2=value2"],
+                "@foo/bar": ["PREBULT_BINARY=http://downloadurl"],
+            }
+            ```
+
+        lifecycle_hooks_execution_requirements: Execution requirements applied to the preinstall, install and postinstall
+            lifecycle hooks on npm packages.
+
+            The execution requirements can be defined per package by package name or globally using "*".
+
+            For example:
+
+            ```
+            lifecycle_hooks_execution_requirements: {
+                "*": ["requires-network"],
+                "@foo/bar": ["no-sandbox"],
+            }
+            ```
+
+        bins: Binary files to create in `node_modules/.bin` for packages in this lock file.
```

Comment:
Is it worth specifying that the paths are relative to the package directory and not node_modules or node_modules/.bin? Maybe that's obvious enough 🤷

Reply:
Should be obvious enough; it is also mentioned that the values are derived from the package.json of the npm package.